CONTRIBUTING.md

```diff
@@ -1,5 +1,90 @@
+
+When proposing a feature we ask that you include:
+
+- A description of the feature to add
+- Any relevant Github issues, pull requests, blog posts, and/or documentation that provides greater context
+
+## Pull Requests
+
+Your pull request should have a description of what it accomplishes. If applicable, PRs should include testing that validate correctness. In all cases the existing tests must pass.
+
+### Guidelines
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+- Search GitHub for an open or closed PR that relates to your submission. You don't want to duplicate existing efforts.
+- Be sure that an existing issue describes the problem you're fixing, or documents the design for the feature you'd like to add. Discussing the design upfront helps to ensure that we're ready to accept your work. Make sure that this issue is referenced in your PR– this can be done by including `Closes: #[issue number]` in the PR's description. Please note that our team may need a week or more before they can review the details in your issue and discuss whether a feature request is something we see value in adding, so bear that in mind before starting work on large PRs.
+- Please sign our Contributor License Agreement (CLA) before sending PRs. We cannot accept code without a signed CLA. Make sure you author all contributed Git commits with the email address associated with your CLA signature.
+
+### Pull Request Lifecycle
+
+1. You are welcome to submit an initial draft pull request for commentary before it is fully completed. It's also helpful to include comments on items you'd like feedback on or feel needs further discussion. Once you believe your pull request is ready to be merged you can set your pull request to open.
+2. When time permits and all checks have passed, Terraform CDK team members will review your PR. From here the pull request will either be merged, or additional changed may be requested in comments. We may also have questions that we need answered about the code, due to something that needs greater clarification or just because we want to better understand your thought process. Please note that the responsibility for passing tests belongs to the author of the PR. If you're having significant trouble with getting tests to pass, please write `@hashicorp/cdktf` in a comment on the PR to request assistance.
+3. When we request changes, you have two options. You can either make those changes or, if you disagree with the suggested changes, a conversation can be had about our respective reasonings where we can then agree on a path forward. In many instances this may be a multi-step process. Pull requests are a great venue for the team and our community to collaborate, and we welcome conversations about how to improve things.
+4. Once all outstanding comments and checklist items have been addressed, your contribution will be merged!
```

Comment:
There is an unmentioned end state which is that the PR is NOT merged. This is often the result of a design that does not fit with the overall product design philosophy or a design that does not fit with the overall product strategy (compatibility with the paid products roadmap). It might be worth finding a way to caveat, probably in Step 2, that "if it is possible to move forward with a review", then the subsequent steps will happen.

Reply:
Great point! Adding extra emphasis around the need to discuss the work to be done in the section above about guidelines might suit this aspect well. Hopefully helping to avoid such instances completely. Nonetheless I like mentioning the not-merged end state where you describe to hit the point home. Thank you for the review!